tool/plugin/lib/plugin.dart

```diff
@@ -657,20 +753,57 @@ class DeployCommand extends ProductCommand {
 /// Generate the plugin.xml from the plugin.xml.template file.
 /// If the --release argument is given, create a git branch and
 /// commit the new file to it, assuming the release checks pass.
-class GenCommand extends Command {
+/// Note: The product-matrix.json file includes a build spec for the
+/// EAP version at the end. When the EAP version is released that needs
+/// to be updated.
+class GenCommand extends ProductCommand {
   final BuildCommandRunner runner;
 
   GenCommand(this.runner);
 
   String get description =>
-      'Generate a valid plugin.xml and .travis.yaml for the Flutter plugin.\n'
+      'Generate a valid plugin.xml and .travis.yml for the Flutter plugin.\n'
       'The plugin.xml.template and product-matrix.json are used as input.';
 
   String get name => 'gen';
 
-  Future<int> run() async {
-    // TODO(messick): Implement GenCommand.
-    throw 'unimplemented';
+  Future<int> doit() async {
+    List json = readProductMatrix();
+    var spec = new SyntheticBuildSpec.fromJson(json.first, release, json.last);
+    log('writing pluginl.xml');
+    int value = 1;
+    var result = await genPluginXml(spec, 'resources');
+    if (result != null) {
+      log('writing .travis.yml');
+      genTravisYml(specs);
+      value = 0;
+    }
+    if (isReleaseMode) {
+      await createGitBranch(release);
+    }
+    return new Future(() => value);
+  }
+
+  Future<int> createGitBranch(String relNumber) async {
+    // This is only called by the gen command when run in release mode,
+    // which means there were no uncommitted changes before gen ran.
+    // TODO(messick) Create a git branch named "release_$relNumber".
+    // Add modified files to the branch then commit it.
+    // The files are:
+    //   resources/META-INF/plugin.xml
+    //   .travis.yml
+    // Testing and debugging is likely to destroy the git repository
+    // so be sure to have a backup.
+    var gitDir = await GitDir.fromExisting(rootPath);
+    var branch = await gitDir.getCurrentBranch();
+    if (branch.branchName != "release_$relNumber") {
+
```

Comment:
Is this block meant to be empty?  (Maybe a missing TODO?)

Reply:
The TODO is above it. The whole method is basically unimplemented.

Reply:
👍